### ISSUE TITLE
No Video after Rebuild on unstable

### ISSUE THREAD

**Issue by j0hax** [NixOS/nixpkgs#116619]
**Describe the bug**
After rebuilding with `nixpkgs-unstable`, I'm stuck at a blurry but functional console. Switching channels (or a flake's nixpkgs URL) to `nixos-20.09` and rebuilding, even without a reboot, has SDDM and Plasma working as expected again (presumably as SDDM gets restarted with different version). 

**To Reproduce**
1. Use nixpkgs-unstable, SDDM, and an AMD GPU
2. Rebuild
3. Reboot

**Expected behavior**
Have correctly-set resolution and a GUI at TTY7

**Additional context**
Running `nixos-rebuild` at some point in the past 3 weeks seems to have broken Video on my AMD R9 290X GPU. Before, the systemd boot text would at some point adapt to my screen resolution during boot, and SDDM would show afterwards. I'm also missing TTY7. Running `systemctl-status display-manager.serivce` reports SDDM is running without a complaint.

I'm not sure which package is exactly causing trouble, as it could be a variety of things going wrong.

**Metadata**
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.1-zen1, NixOS, 20.09.20210316.36e15cd (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210308_1c0e3e4`
 - channels(root): `"nixos-21.05pre276379.266dc8c3d05"`
 - channels(johannes): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: [
    services.xserver.sddm
]
```

**Comment by j0hax** [comment]
Problem has been resolved by manually setting `boot.kernelParams = [ "radeon.cik_support=0" "amdgpu.cik_support=1" ];` in configuration.nix as per the [wiki](https://nixos.wiki/wiki/AMD_GPU).

Strangely my graphics always worked without it being set before, but as everything is working now, I'll close.

**Comment by Mathnerd314** [comment]
I had black screens on sddm startup; there was a kernel null pointer deref in the logs. I'm using dual graphics, RX 570 (radeonsi) + onboard AMD graphics (radeon). It seems to be fixed in 5.11.7 as in this bug: https://bugzilla.kernel.org/show_bug.cgi?id=212137
